crates/biome_json_analyze/src/lint/nursery/use_required_scripts.rs

```diff
@@ -0,0 +1,143 @@
+use biome_analyze::{Ast, Rule, RuleDiagnostic, context::RuleContext, declare_lint_rule};
+use biome_console::markup;
+use biome_json_syntax::JsonRoot;
+use biome_rowan::{AstNode, AstSeparatedList};
+use biome_rule_options::use_required_scripts::UseRequiredScriptsOptions;
+
+declare_lint_rule! {
+    /// Enforce the presence of required scripts in package.json.
+    ///
+    /// This rule ensures that specified scripts are defined in the `scripts` section of a `package.json` file.
+    /// It's particularly useful in monorepo environments where consistency across workspaces is important.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "requiredScripts": ["test", "build"]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ```json,expect_diagnostic,use_options
+    /// {
+    ///     "scripts": {
+    ///         "test": "vitest"
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```json,use_options
+    /// {
+    ///     "scripts": {
+    ///         "test": "vitest",
+    ///         "build": "tsc"
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ## Options
+    ///
+    /// ### `requiredScripts`
+    ///
+    /// An array of script names that must be present in the `scripts` section of `package.json`.
+    /// Default: `[]` (no scripts required)
+    ///
+    pub UseRequiredScripts {
+        version: "next",
+        name: "useRequiredScripts",
+        language: "json",
+        recommended: false,
+    }
+}
+
+const PACKAGE_JSON: &str = "package.json";
+
+impl Rule for UseRequiredScripts {
+    type Query = Ast<JsonRoot>;
+    type State = Vec<String>;
+    type Signals = Option<Self::State>;
+    type Options = UseRequiredScriptsOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let query = ctx.query();
+        let path = ctx.file_path();
+        let options = ctx.options();
+
+        if !path.ends_with(PACKAGE_JSON) {
+            return None;
+        }
+        if options.required_scripts.is_empty() {
+            return None;
+        }
```

Comment:
This is another bug, I believe. The rule should enforce the scripts that belong to this option. If this list is empty **but we have scripts**, we must raise an error. What do you think? Otherwise the rule is completely useless without options

Reply:
is it possible to enforce in the biome config that the list is non-empty when the rule is defined? Without requiredScripts, the rule means nothing.

Reply:
added & rephrased for this rule 👍